neural/predict_with_neural.py

```diff
@@ -0,0 +1,227 @@
+TO_GPU = True  # NOTE: this does not currently work for False
+GPU_INDEX = 0
+BIOBERT_EXP_DIR = '2019-10-28_15-59-09'
+QUESTION_FOLD_MAP = {
+    4: 0,
+    5: 0,
+    9: 0,
+    10: 0,
+    11: 0,
+}
+
+
+def identify_attended_senteces(docid_attnweights_map, proc_articles_repr, topk=5):
+    attended_sents = return_attnw_over_sents(docid_attnweights_map, proc_articles_repr, topk)
+    return attended_sents
+
+
+def run_predict(q_docpartitions, q_fold_config_map, bertmodel, q_state_dict_path_map, results_dir, sents_embed_dir,
+                question_fold_map, to_gpu, gpu_index, num_epochs=1):
+    q_predictions = {}
+    for question in q_fold_config_map:
+        mconfig, options, __ = q_fold_config_map[question]
+        options['num_epochs'] = num_epochs  # override number of epochs using user specified value
+
+        # update options fold num to the current fold
+        options['fold_num'] = question_fold_map[question]
+        data_partition = q_docpartitions[question][options['fold_num']]
+
+        results_path = os.path.join(results_dir, 'question_{}'.format(question), 'fold_{}'.format(options['fold_num']))
+        results_wrk_dir = create_directory(results_path)
+
+        q_predictions[question] = predict_neural_discern(data_partition, bertmodel, mconfig, options,
+                                                         results_wrk_dir, sents_embed_dir,
+                                                         state_dict_dir=q_state_dict_path_map[question], to_gpu=to_gpu,
+                                                         gpu_index=gpu_index)
+    return q_predictions
+
+
+def retrieve_page_from_internet(url: str):
+    res = requests.get(url)
+    html_page = res.content.decode("utf-8")
+    return html_page
+
+
+def create_prediction_qdoc_partitions(questions: List[int], question_fold_map: Dict[int, int]):
+    q_docpartitions = {}
+    for q in questions:
+        q_docpartitions[q] = {}
+        fold_num = question_fold_map[q]
+        q_docpartitions[q][fold_num] = {'test': [0]}  # output of np.vectorize() ?!
+    return q_docpartitions
+
+
+def build_DataDictProcessor(data_dict, vocab_path: str, processor_config: Dict):
+    # from the first notebook
+
+    processor = DataDictProcessor(processor_config)
+    processor.generate_articles_repr(data_dict)
+    return processor
+
+
+def embed_sentences(docs_data_tensor, sents_embed_path, bertmodel, bert_config, gpu_index):
+    # from the second notebook
+
+    bertembeder = BertEmbedder(bertmodel, bert_config)
+    fdtype = torch.float32
+
+    # generate and dump bert embedding for the tokens inside the specificed embedding directory
+    bert_proc_docs = generate_sents_embeds_from_docs(docs_data_tensor, bertembeder, sents_embed_path, fdtype,
+                                                     gpu_index=gpu_index)
+    ReaderWriter.dump_data(bert_proc_docs, os.path.join(sents_embed_path, 'bert_proc_docs.pkl'))
+
+    return sents_embed_path
```

Comment:
i am not sure why you would need to return `sents_embed_path` if you are already passing it `argument` to the function.